icicle/cmake/Common.cmake

```diff
@@ -14,10 +14,12 @@ endfunction()
 function(set_gpu_env)
     # add the target cuda architectures
     # each additional architecture increases the compilation time and output file size
-    if(${CMAKE_VERSION} VERSION_LESS "3.24.0")
-    set(CMAKE_CUDA_ARCHITECTURES ${CUDA_ARCH} PARENT_SCOPE)
+	if(DEFINED CUDA_ARCH) # user defined arch takes priority
+	    set(CMAKE_CUDA_ARCHITECTURES ${CUDA_ARCH})
```

Comment:
did it work for you? why did you remove the PARENT_SCOPE? is it not needed outside this function?
When I print it in the main cmake I get the wrong version:
message("CMAKE_CUDA_ARCHITECTURES=${CMAKE_CUDA_ARCHITECTURES}")

Are you sure it compiled the correct version?

Reply:
The PARENT_SCOPE indeed should be added. Thanks for your review.